### ISSUE TITLE
[1.13.0-rc7] panic: close of nil channel

### ISSUE THREAD

**Issue by JamesJJ** [docker/docker#30659]
**Description**

`Jan 31 00:25:41 ip-10-12-76-243 dockerd[5328]: panic: close of nil channel`

**Describe the results you received:**

```
[root@ip-10-12-76-243 centos]# journalctl -u docker.service
-- Logs begin at Mon 2017-01-23 11:01:58 UTC, end at Thu 2017-02-02 01:53:02 UTC. --
Jan 23 11:03:12 ip-10-12-76-243 systemd[1]: Starting Docker Application Container Engine...
Jan 23 11:03:19 ip-10-12-76-243 dockerd[5328]: time="2017-01-23T11:03:19.069893728Z" level=info msg="libcontainerd: new containerd process, pid: 5372"
Jan 23 11:03:20 ip-10-12-76-243 dockerd[5328]: time="2017-01-23T11:03:20.114209259Z" level=info msg="devmapper: Creating filesystem xfs on device docker-202:1-17039471-base"
Jan 23 11:03:20 ip-10-12-76-243 dockerd[5328]: time="2017-01-23T11:03:20.439029859Z" level=info msg="devmapper: Successfully created filesystem xfs on device docker-202:1-17039471
Jan 23 11:03:20 ip-10-12-76-243 dockerd[5328]: time="2017-01-23T11:03:20.552139118Z" level=info msg="Graph migration to content-addressability took 0.00 seconds"
Jan 23 11:03:20 ip-10-12-76-243 dockerd[5328]: time="2017-01-23T11:03:20.552942042Z" level=info msg="Loading containers: start."
Jan 23 11:03:22 ip-10-12-76-243 dockerd[5328]: time="2017-01-23T11:03:22.228113573Z" level=info msg="Firewalld running: false"
Jan 23 11:03:22 ip-10-12-76-243 dockerd[5328]: time="2017-01-23T11:03:22.631498507Z" level=info msg="Default bridge (docker0) is assigned with an IP address 172.17.0.0/16. Daemon 
Jan 23 11:03:22 ip-10-12-76-243 dockerd[5328]: time="2017-01-23T11:03:22.696410494Z" level=info msg="Loading containers: done."
Jan 23 11:03:23 ip-10-12-76-243 dockerd[5328]: time="2017-01-23T11:03:23.123110886Z" level=info msg="Daemon has completed initialization"
Jan 23 11:03:23 ip-10-12-76-243 dockerd[5328]: time="2017-01-23T11:03:23.123142981Z" level=info msg="Docker daemon" commit=48a9e53 graphdriver=devicemapper version=1.13.0-rc7
Jan 23 11:03:23 ip-10-12-76-243 dockerd[5328]: time="2017-01-23T11:03:23.135151802Z" level=info msg="API listen on /var/run/docker.sock"
Jan 23 11:03:23 ip-10-12-76-243 systemd[1]: Started Docker Application Container Engine.
Jan 23 11:05:57 ip-10-12-76-243 dockerd[5328]: time="2017-01-23T11:05:57.013656966Z" level=error msg="Handler for GET /v1.22/networks/cmsservice_default returned error: network cm
Jan 23 11:06:02 ip-10-12-76-243 dockerd[5328]: time="2017-01-23T11:06:02Z" level=info msg="Firewalld running: false"
Jan 23 11:06:04 ip-10-12-76-243 dockerd[5328]: time="2017-01-23T11:06:04Z" level=info msg="Firewalld running: false"
Jan 23 11:06:06 ip-10-12-76-243 dockerd[5328]: time="2017-01-23T11:06:06Z" level=info msg="Firewalld running: false"
Jan 23 11:06:06 ip-10-12-76-243 dockerd[5328]: time="2017-01-23T11:06:06Z" level=info msg="Firewalld running: false"
Jan 31 00:25:24 ip-10-12-76-243 dockerd[5328]: time="2017-01-31T00:23:29.137459513Z" level=warning msg="Connect failed: dial udp 10.12.0.2:53: i/o timeout"
Jan 31 00:25:24 ip-10-12-76-243 dockerd[5328]: time="2017-01-31T00:25:23.508012062Z" level=warning msg="Connect failed: dial udp 10.12.0.2:53: i/o timeout"
Jan 31 00:25:25 ip-10-12-76-243 dockerd[5328]: time="2017-01-31T00:25:25.516158085Z" level=error msg="containerd: deleting container" error="exit status 1: \"container 0b31c9d92dc
Jan 31 00:25:26 ip-10-12-76-243 dockerd[5328]: time="2017-01-31T00:25:26.586898742Z" level=warning msg="Health check for container bcdbc8abd8fdbc7044580344b863d652280f187d466bd453
Jan 31 00:25:27 ip-10-12-76-243 dockerd[5328]: time="2017-01-31T00:25:27Z" level=info msg="Firewalld running: false"
Jan 31 00:25:41 ip-10-12-76-243 dockerd[5328]: time="2017-01-31T00:25:41.443917791Z" level=error msg="libcontainerd: failed to receive event from containerd: rpc error: code = 13 
Jan 31 00:25:41 ip-10-12-76-243 dockerd[5328]: time="2017-01-31T00:25:41.504208325Z" level=info msg="libcontainerd: new containerd process, pid: 13868"
Jan 31 00:25:41 ip-10-12-76-243 dockerd[5328]: panic: close of nil channel
Jan 31 00:25:41 ip-10-12-76-243 dockerd[5328]: goroutine 30 [running]:
Jan 31 00:25:41 ip-10-12-76-243 dockerd[5328]: panic(0x8078a0, 0xc4202325c0)
Jan 31 00:25:41 ip-10-12-76-243 dockerd[5328]: /usr/local/go/src/runtime/panic.go:500 +0x1a1
Jan 31 00:25:41 ip-10-12-76-243 dockerd[5328]: github.com/docker/containerd/supervisor.(*Supervisor).execExit.func1(0xc420115d60, 0xc42010b520, 0x0)
Jan 31 00:25:41 ip-10-12-76-243 dockerd[5328]: /go/src/github.com/docker/containerd/supervisor/exit.go:90 +0x10c
Jan 31 00:25:41 ip-10-12-76-243 dockerd[5328]: created by github.com/docker/containerd/supervisor.(*Supervisor).execExit
Jan 31 00:25:41 ip-10-12-76-243 dockerd[5328]: /go/src/github.com/docker/containerd/supervisor/exit.go:91 +0xee
Jan 31 00:25:42 ip-10-12-76-243 dockerd[5328]: time="2017-01-31T00:25:42.076459174Z" level=info msg="libcontainerd: new containerd process, pid: 13874"
Jan 31 00:25:47 ip-10-12-76-243 dockerd[5328]: time="2017-01-31T00:25:47.962587644Z" level=error msg="Handler for GET /containers/json returned error: write unix /var/run/docker.s
Jan 31 00:25:47 ip-10-12-76-243 dockerd[5328]: http: multiple response.WriteHeader calls
Jan 31 00:26:03 ip-10-12-76-243 dockerd[5328]: http: multiple response.WriteHeader calls
Jan 31 00:26:03 ip-10-12-76-243 dockerd[5328]: time="2017-01-31T00:26:03.867046212Z" level=error msg="Handler for GET /containers/json returned error: write unix /var/run/docker.s
Jan 31 00:26:07 ip-10-12-76-243 dockerd[5328]: time="2017-01-31T00:26:07.139583343Z" level=warning msg="Health check for container bcdbc8abd8fdbc7044580344b863d652280f187d466bd453
Feb 01 04:41:05 ip-10-12-76-243 dockerd[5328]: time="2017-02-01T04:41:05.140882775Z" level=warning msg="containerd: unable to save bcdbc8abd8fdbc7044580344b863d652280f187d466bd453
Feb 01 14:54:01 ip-10-12-76-243 dockerd[5328]: time="2017-02-01T14:54:01.716693438Z" level=warning msg="Connect failed: dial udp 10.12.0.2:53: i/o timeout"
Feb 01 14:54:01 ip-10-12-76-243 dockerd[5328]: time="2017-02-01T14:54:01.904030673Z" level=error msg="libcontainerd: failed to receive event from containerd: rpc error: code = 13 
Feb 01 14:54:01 ip-10-12-76-243 dockerd[5328]: time="2017-02-01T14:54:01.924336279Z" level=info msg="libcontainerd: new containerd process, pid: 16638"
Feb 01 14:54:02 ip-10-12-76-243 dockerd[5328]: time="2017-02-01T14:54:02.083205444Z" level=info msg="libcontainerd: new containerd process, pid: 16639"
Feb 01 15:47:50 ip-10-12-76-243 dockerd[5328]: time="2017-02-01T15:47:48.523888373Z" level=info msg="libcontainerd: new containerd process, pid: 1784"
Feb 01 15:47:51 ip-10-12-76-243 dockerd[5328]: panic: close of nil channel
Feb 01 15:47:51 ip-10-12-76-243 dockerd[5328]: goroutine 33 [running]:
Feb 01 15:47:51 ip-10-12-76-243 dockerd[5328]: panic(0x8078a0, 0xc42020b720)
Feb 01 15:47:51 ip-10-12-76-243 dockerd[5328]: /usr/local/go/src/runtime/panic.go:500 +0x1a1
Feb 01 15:47:51 ip-10-12-76-243 dockerd[5328]: github.com/docker/containerd/supervisor.(*Supervisor).execExit.func1(0xc4202a4320, 0xc4200f9790, 0x0)
Feb 01 15:47:51 ip-10-12-76-243 dockerd[5328]: /go/src/github.com/docker/containerd/supervisor/exit.go:90 +0x10c
Feb 01 15:47:51 ip-10-12-76-243 dockerd[5328]: created by github.com/docker/containerd/supervisor.(*Supervisor).execExit
Feb 01 15:47:51 ip-10-12-76-243 dockerd[5328]: /go/src/github.com/docker/containerd/supervisor/exit.go:91 +0xee
Feb 01 15:47:51 ip-10-12-76-243 dockerd[5328]: time="2017-02-01T15:47:51.509274026Z" level=error msg="libcontainerd: failed to receive event from containerd: rpc error: code = 13 
```

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client:
 Version:      1.13.0-rc7
 API version:  1.25
 Go version:   go1.7.3
 Git commit:   48a9e53
 Built:        Fri Jan 13 06:50:32 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.13.0-rc7
 API version:  1.25 (minimum version 1.12)
 Go version:   go1.7.3
 Git commit:   48a9e53
 Built:        Fri Jan 13 06:50:32 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Output of `docker info`:**

```
Containers: 4
 Running: 4
 Paused: 0
 Stopped: 0
Images: 4
Server Version: 1.13.0-rc7
Storage Driver: devicemapper
 Pool Name: docker_lvm_vg-docker_lvm
 Pool Blocksize: 524.3 kB
 Base Device Size: 8.59 GB
 Backing Filesystem: xfs
 Data file: 
 Metadata file: 
 Data Space Used: 2.705 GB
 Data Space Total: 49.92 GB
 Data Space Available: 47.22 GB
 Metadata Space Used: 852 kB
 Metadata Space Total: 1.615 GB
 Metadata Space Available: 1.614 GB
 Thin Pool Minimum Free Space: 3.494 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: true
 Deferred Deleted Device Count: 0
 Library Version: 1.02.135-RHEL7 (2016-09-28)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: N/A (expected: 03e5862ec0d8d3b3f750e19fca3ee367e13c090e)
runc version: 2f7393a47307a16f8cee44a37b262e8b81021e3e
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 3.10.0-514.2.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 1.795 GiB
Name: ip-10-12-76-243
ID: SGKM:ARMQ:A4MR:HMLP:CSP7:QOKC:Q52E:6CJO:7DJQ:URHR:S6DC:2D6R
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

* Centos 7, fully updated
* AWS / EBS Disk
* Docker from `https://yum.dockerproject.org/repo/testing/centos/7`

**Comment by vieux** [comment]
@mlaventure this one is fixed in 1.13.x already right ?

**Comment by tonistiigi** [comment]
@vieux Yes, this is same as #30469
